### PR TITLE
Miniapp Script Improvements

### PR DESCRIPTION
**Pull request by tenbrugg**
including cleanup, simpler thread count handling and
improved time-out handling.

Signed-off-by: Monika ten Bruggencate tenbrugg@gmail.com

@sungeunchoi 
Relates to #12

### CODE REVIEW

File: scripts/miniapps/run_miniapps [hunk 1]
```diff
@@ -107,9 +118,24 @@ else
...
-    timeout_2="-t 1200"
-    timeout_3="-t 1500"
+    timeout_2="-t 1800"
+    timeout_3="-t 2400"
+fi
+
+if [ $size = "small" ]; then
+    timeout=$timeout_1
+else if [ $size = "medium" ]; then
+    timeout=$timeout_1
```
Comment (sungeunchoi) [inline scripts/miniapps/run_miniapps:128]: Is this supposed to be `timeout_2`?

**Comment by tenbrugg** [comment]
No, timeout_1 works for small and medium, timeout_2 is a placeholder to be used some day when MPICH Finalize is faster and you guys can lower the time out value. I did some experiments and timeout_2 should work then.

**Comment by sungeunchoi** [comment]
Oh now I see.   To avoid confusion, can you rename `timeout_3` to `timeout_large` and also rename `timeout_2` to `timeout_large_2`.  You could probably also comment out the assignment to `timeout_large_2`, since it's not being used.

**Comment by sungeunchoi** [comment]
👍 after discussion.

**Pull request merged**